magda-web-client/src/UI/Tooltip.js

```diff
@@ -1,24 +1,110 @@
 import "./Tooltip.css";
 
 import React from "react";
-import { Link } from "react-router-dom";
-
-import helpIcon from "../assets/help-24.svg";
 
 /**
  * @description Return a information tooltip, on hover show calculation method.
  * @returns { div }
  */
-const Tooltip = () => {
-    return (
-        <div className="tooltip">
-            <img src={helpIcon} alt="Help Link" />
-            <span className="tooltiptext">
-                Calculated using the{" "}
-                <Link to="/page/dataset-quality">Open data scale</Link>
-            </span>
-        </div>
-    );
-};
+class Tooltip extends React.Component {
+    state = {
+        offset: 0,
+        dismissed: false
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.rootRef = React.createRef();
+        this.tooltipTextElementRef = React.createRef();
+    }
+
+    componentDidMount() {
+        if (this.props.requireClickToDismiss) {
+            document.addEventListener("mousedown", this.handleClickOutside);
+        }
+        this.adjustOffset();
+    }
+
+    componentDidUpdate() {
+        this.adjustOffset();
```

Comment:
change state here cause an infinite loop, eg, if you go to http://localhost:6108/dataset/ds-dap-csiro%3A17050/details?q=water and wait for a few seconds